plugin/trino-hive/src/main/java/io/trino/plugin/hive/HdfsEnvironment.java

```diff
@@ -45,6 +45,7 @@
     private final Optional<FsPermission> newDirectoryPermissions;
     private final boolean newFileInheritOwnership;
     private final boolean verifyChecksum;
+    private static ThreadLocal<ConnectorIdentity> connectorIdentity = new ThreadLocal<>();
```

Comment:
You should use io.trino.plugin.hive.s3.S3SecurityMappingConfigurationProvider. Or something similar because I am not sure if it fits the `SecurityMapping` scope. Surely `ThreadLocal` is not a good way go.

Reply:
I too did not like `ThreadLocal` but was not sure how to do this. Thanks for the pointer to `S3SecurityMappingConfigurationProvider`.

I pushed an update to use `S3SecurityMappingConfigurationProvider`. With this update, if a user does not provide a `hive.s3.security-mapping.config-file` file then they can change the role session name but they cannot have it include current session user name. 

I think this is OK and provides more flexibility as different role session names can be used for different IAM roles.